### DISCUSSION TITLE
How can I prevent S3 API calls when deleting records?

### DISCUSSION THREAD

**Discussion by jlak0** [pocketbase/pocketbase#3991] [Q&A] [answered]
I'm using Cloudflare R2 as S3 storage. When I delete records, my PB instance calls the R2 API twice (once for Class A Operations and once for Class B Operations). I tested it in a newly installed PB instance, with a collection that has only one text field (without a file field). When deleting a lot, it may cause unnecessary expenses.

**Comment by ganigeorgiev** [comment, marked as answer]
At the moment there is no hook to intercept only the files delete operation (eventually will be added as part of https://github.com/pocketbase/pocketbase/issues/2387).

PocketBase performs two types of delete operations:
- direct delete - when a single file entry from a record is deleted
- delete by a prefix - when an entire record or collection is deleted to delete all related files to the record/collection

Depending on your use case, one possible workaround could be to use soft-deletion, aka. instead of deleting the record add a `bool` field `deleted`/`archived` and change its state.
